chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -72,36 +231,66 @@ void MdTextButtonBase::UpdateColorsForBrave() {
   }
 }
 
-MdTextButton::MdTextButton(PressedCallback callback,
-                           const std::u16string& text,
-                           int button_context)
-    : MdTextButtonBase(std::move(callback), text, button_context) {
-  SetCornerRadius(100);
-  views::HighlightPathGenerator::Install(
-      this, std::make_unique<BraveTextButtonHighlightPathGenerator>());
-  auto* ink_drop = views::InkDrop::Get(this);
-  views::InkDrop::UseInkDropForFloodFillRipple(ink_drop,
-                                               /*highlight_on_hover=*/false,
-                                               /*highlight_on_focus=*/true);
-  ink_drop->SetCreateHighlightCallback(base::BindRepeating(
-      [](Button* host) {
-        const SkColor fill_color = SK_ColorTRANSPARENT;
-        gfx::RectF boundsF(host->GetLocalBounds());
-        return std::make_unique<InkDropHighlight>(
-            boundsF.size(), static_cast<MdTextButton*>(host)->GetCornerRadius(),
-            boundsF.CenterPoint(), fill_color);
-      },
-      this));
-}
+// To be called from MdTextButtonBase::UpdateColors().
+void MdTextButton::UpdateColorsForBrave() {
+  if (GetKind() == Kind::kOld) {
+    UpdateOldColorsForBrave();
+    return;
+  }
 
-MdTextButton::~MdTextButton() = default;
+  // Leo buttons only have a light and dark mode.
+  auto color_scheme =
+      GetNativeTheme()->GetPreferredColorScheme() == ColorScheme::kDark
+          ? ColorScheme::kDark
+          : ColorScheme::kLight;
+  auto state = GetVisualState();
+  float opacity = 1;
 
-SkPath MdTextButton::GetHighlightPath() const {
-  SkPath path;
-  int radius = GetCornerRadius();
-  path.addRRect(
-      SkRRect::MakeRectXY(RectToSkRect(GetLocalBounds()), radius, radius));
-  return path;
+  // Leo buttons don't have a pressed state, so use the hover state instead.
+  if (state == ButtonState::STATE_PRESSED) {
+    state = ButtonState::STATE_HOVERED;
+  }
+
+  // The loading style is the normal button style, with some opacity.
+  if (loading_) {
+    state = ButtonState::STATE_NORMAL;
+    opacity = kLoadingOpacity;
+  }
+
+  // The enabled style is the normal button style with more opacity.
+  if (!GetEnabled() || state == STATE_DISABLED) {
+    state = ButtonState::STATE_DISABLED;
+    opacity = kDisabledOpacity;
+  }
+
+  MdTextButtonStyleKey style_lookup{GetKind(), color_scheme, state};
+  auto it = g_button_themes.find(style_lookup);
+  if (it == g_button_themes.end()) {
+    NOTREACHED() << "No style found for ButtonKind: " << kind_
+                 << ", ColorScheme: "
+                 << (color_scheme == ColorScheme::kDark ? "dark" : "light")
+                 << ", ButtonState: " << state;
+  }
+  auto style = it->second;
```

Comment:
nit: avoid copying with const & could be possible
```suggestion
  const auto& style = it->second;
```

Reply:
That's my non-C++ background shining through :laughing: